server/server.go

```diff
@@ -392,15 +394,27 @@ func (s *Server) handleReq(conn *gokeyless.Conn, ch chan *gokeyless.Packet) {
 }
 
 // Serve accepts incoming connections on the Listener l, creating a new service goroutine for each.
-func (s *Server) Serve(l net.Listener, timeout time.Duration) error {
+func (s *Server) Serve(l net.Listener) error {
 	defer l.Close()
+
+	ch := make(chan req, 8)
+	for i := 0; i < 8; i++ {
+		go s.handleReqs(ch)
+	}
+	defer func() {
+		s.workerMu.Lock()
+		s.closed = true
+		close(ch)
+		s.workerMu.Unlock()
+	}()
+
 	for {
 		c, err := l.Accept()
 		if err != nil {
 			log.Error(err)
 			return err
 		}
-		go s.handle(gokeyless.NewConn(tls.Server(c, s.Config)), timeout)
+		go s.handle(gokeyless.NewConn(tls.Server(c, s.Config)), ch)
```

Comment:
so I believe this was causing goroutine to grow out of control, since each `go s.handle()` will spawn a goroutine which is blocked for one hour for each incoming connection. And when a lot of test machines trying to do just one key signing request, the server just has a lot of goroutine waiting for nothing.  The original fix, arguably ugly, is to has a smaller timeout for tcp connection, so "we hope" some goroutine will exit naturally after timeout. Frankly, I do feel that won't help much.

So in order to use the channel model so we don't create unlimited number of goroutine, I think we need to have this master thread block waiting for a available worker,  so, simply stuffing new (conn, pakcket) into a channel will do that, and we can eliminate this `go s.handle call()`. What do you think?